File: RxBinding/app/src/main/java/com/safframework/study/rxbinding/utils/RxUtils.java
package com.safframework.study.rxbinding.utils;

import android.support.v7.app.AppCompatActivity;

import com.safframework.lifecycle.RxLifecycle;
import com.safframework.utils.RxJavaUtils;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by tony on 2017/9/18.
 */

public class RxUtils {

    /**
     * 对RxView绑定的事件
     * 封装了防止重复点击和RxLifecycle的生命周期
     */
    public static ObservableTransformer useRxViewTransformer(final AppCompatActivity targetActivity) {

        return new ObservableTransformer() {

            @Override
            public ObservableSource apply(Observable upstream) {
                return upstream.compose(RxJavaUtils.preventDuplicateClicksTransformer())
                        .compose(RxLifecycle.bind(targetActivity).toLifecycleTransformer());
            }
        };
    }
}

File: StudyOperator/app/src/main/java/com/safframework/study/operator/utils/RxUtils.kt
package com.safframework.study.operator.utils

import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import com.safframework.lifecycle.RxLifecycle
import com.safframework.utils.RxJavaUtils
import io.reactivex.ObservableTransformer

/**
 * Created by Tony Shen on 2017/7/20.
 */
object RxUtils {

    /**
     * 对RxView绑定的事件
     * 封装了防止重复点击和RxLifecycle的生命周期
     */
    @JvmStatic
    fun <T> useRxViewTransformer(targetActivity: AppCompatActivity): ObservableTransformer<T, T> {

        return ObservableTransformer { upstream ->
            upstream.compose(RxJavaUtils.preventDuplicateClicksTransformer())
                    .compose(RxLifecycle.bind(targetActivity).toLifecycleTransformer())
        }
    }

    /**
     * 对RxView绑定的事件
     * 封装了防止重复点击和RxLifecycle的生命周期
     */
    @JvmStatic
    fun <T> useRxViewTransformer(targetFragment: Fragment): ObservableTransformer<T, T> {

        return ObservableTransformer { upstream ->
            upstream.compose(RxJavaUtils.preventDuplicateClicksTransformer())
                    .compose(RxLifecycle.bind(targetFragment).toLifecycleTransformer())
        }
    }

    @JvmStatic
    fun <T> routerUriTransformer(textView: TextView): ObservableTransformer<T, String> {

        return ObservableTransformer { upstream ->
            upstream.map {
                it -> textView.getText().toString().toLowerCase() + "/" + textView.getText().toString()
            }
        }
    }
}